Dockerfile

```diff
@@ -39,31 +36,158 @@ RUN conda install --name python2 --quiet --yes \
         /home/jovyan/.local/share/jupyter/kernels/python2/kernel.json
 
-# Don't rename user- causes problems with earlier hardcoded paths
-#RUN usermod -l omero jovyan -m -d /home/omero
-#USER omero
+# Install git and pull the notebooks from the training repository
+RUN conda install --name python2 --quiet --yes -c anaconda git
+
+# Switch to root user for installing Cell Profiler dependencies
+USER root
+
+# Install CellProfiler dependencies
+RUN   apt-get -y update &&                                          \
+      apt-get -y install                                            \
+        build-essential    \
+        cython             \
+        git                \
+        libmysqlclient-dev \
+        libhdf5-dev        \
+        libxml2-dev        \
+        libxslt1-dev       \
+        openjdk-8-jdk      \
+        python-dev         \
+        python-pip         \
+        python-h5py        \
+        python-matplotlib  \
+        python-mysqldb     \
+        python-scipy       \
+        python-numpy       \
+        python-wxgtk3.0    \
+        python-zmq
+
+WORKDIR /usr/local/src
+
+# Install CellProfiler
+RUN /opt/conda/envs/python2/bin/git clone https://github.com/CellProfiler/CellProfiler.git
+
+WORKDIR /usr/local/src/CellProfiler
+
+ARG version=3.0.0
+
+RUN /opt/conda/envs/python2/bin/git checkout tags/v$version
+
+RUN /opt/conda/envs/python2/bin/pip install --editable .
+
+# R Installation (Staying as root user)
+RUN apt-get update -y && \
+    apt-get install -y nodejs wget git
+
+## Install R tools
+RUN /opt/conda/envs/python2/bin/conda config --add channels bioconda && \
+    /opt/conda/envs/python2/bin/conda install --quiet --yes \
+    'r-mclust' \
+    'r-ggdendro' \
+    'r-igraph' \
+    'r-pheatmap'
 
+# Dependencies necessary for install.R
+RUN echo "deb-src http://deb.debian.org/debian testing main" >> /etc/apt/sources.list
+RUN echo "deb http://ftp.debian.org/debian jessie-backports main" >> /etc/apt/sources.list
+RUN apt-get update && \
+    apt-get -y install libssl-dev libxml2-dev libcurl4-openssl-dev
+
+#########################################################################################
+## Necessary for running maven
+## copied from https://hub.docker.com/r/cardcorp/r-java/~/dockerfile/
+##
+## gnupg is needed to add new key
+RUN apt-get update && apt-get install -y gnupg2
+
+## Install Java 
+RUN apt-get update \
+    && apt-get install -t jessie-backports -y openjdk-8-jdk \
+    && echo "MAVEN IS NOT IN THE UPSTREAM LIST (JOSH)" \
+    && apt-get install -y maven \ 
+    && rm -rf /var/lib/apt/lists/* \
+    && apt-get clean 
+
+ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+RUN rm -rf /usr/lib/jvm/java
+RUN ln -s  /usr/lib/jvm/java-8-openjdk-amd64 /usr/lib/jvm/java
+
+RUN /opt/conda/envs/python2/bin/conda install --quiet --yes -c r r
+RUN R CMD javareconf
+    
+## make sure Java can be found in rApache and other daemons not looking in R ldpaths
```

Comment:
Is this needed in this image?

Reply:
ref : https://cran.r-project.org/doc/manuals/r-release/R-admin.html

"Binary distributions of R are built against a specific version (e.g. 1.8.0_121) of Oracle Java so sudo R CMD javareconf may be needed before using Java-using packages."